api/v1alpha1/bucket_types.go

```diff
@@ -36,14 +36,18 @@ type BucketSpec struct {
 	// +kubebuilder:validation:Optional
 	Paths []string `json:"paths,omitempty"`
 
+	// s3InstanceRef where create the bucket
+	// +kubebuilder:validation:Optional
+	S3InstanceRef string `json:"s3InstanceRef,omitempty"`
```

Comment:
Create an immutable field to simplify code.

See https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/